src/converse-autocomplete.js

```diff
@@ -40,6 +40,19 @@ converse.plugins.add("converse-autocomplete", {
             const element = document.createElement("li");
             element.setAttribute("aria-selected", "false");
 
+            const img = document.createElement("img");
+            let dataUri = "data:" + _converse.DEFAULT_IMAGE_TYPE + ";base64," + _converse.DEFAULT_IMAGE;
+
+            if (_converse.vcards) {
+                const vcard = _converse.vcards.findWhere({'nickname': text});
+                if (vcard) dataUri = "data:" + vcard.get('image_type') + ";base64," + vcard.get('image');
+            }
+
+            img.setAttribute("src", dataUri);
+            img.setAttribute("width", "22");
+            img.setAttribute("class", "avatar avatar-autocomplete");
+            element.appendChild(img);
+
```

Comment:
This code is too specific to the MUC nickname autocomplete use-case, which is a subset of all different autocomplete implementations, and should therefore not be here. The autocomplete plugin is meant to be generic and we cannot assume that the text values are nicknames or that we want to show avatars for them.

So this code needs to be put into a different function (e.g. MUC_NICKNAME) which can in turn call ITEM and then it should be possible to pass that function in to the `AutoComplete` constructor in the `config` map so that it can be used instead of ITEM (which can stay the default).

Reply:
OK. I shall take another pass at this. Thanks